src/libcore/panic.rs

```diff
@@ -45,11 +45,20 @@ impl<'a> PanicInfo<'a> {
                 issue = "0")]
     #[doc(hidden)]
     #[inline]
-    pub fn internal_constructor(message: Option<&'a fmt::Arguments<'a>>,
-                                location: Location<'a>)
-                                -> Self {
+    pub fn internal_constructor(
+        message: Option<&'a fmt::Arguments<'a>>,
+        location: &'a Location<'a>,
+    ) -> Self {
         struct NoPayload;
-        PanicInfo { payload: &NoPayload, location, message }
+        PanicInfo {
+            message,
+            payload: &NoPayload,
+            location: Location {
```

Comment:
Leave a FIXME comment about potentially making `PanicInfo` hold a reference to `Location`.

Reply:
If you think it should hold a reference, I can make that change now. I had it previously but your question about its behavior with unsafe code made me think to not change the type decl.

Reply:
This is why I want a FIXME comment (and maybe even an issue) - the way it is right now works, and we should maybe investigate later whether it would be better not to copy the `Location` out.

Reply:
https://github.com/rust-lang/rust/issues/65856